### DISCUSSION TITLE
bsp in their own repo

### DISCUSSION THREAD

**Discussion by lowlander** [zephyrproject-rtos/zephyr#28885] [Ideas] [answered]
The Zephyr `boards/*/` dirs are growing and growing (I have two bsp's in the pipeline myself). Would it be an idea to make a `zephyr_boards` repo where we collect all boards, and only have a very limited, maybe only qemu boards,  number of boards in the `zephyr` main repo ?

If not, would it be an idea to atleast make vendor subdirectories? like `boards/arm/st/nucleo*`,  `boards/arm/nxp/*`, `boards/arm/olimex/*`, etc. ?

**Comment by martinjaeger** [comment, marked as answer]
I thought about that aswell. I'm also a bit concerned that the repository size will grow excessively, as each board usually adds at least one picture of a few 100 kB.

But the disadvantage is that it's much more difficult to keep them synchronized with the rest of the Zephyr repo if they are out of tree... maybe you want to open an issue for that? Seems like not many people are looking into the discussions :)

**Reply by lowlander** [reply]
I first wanted to try this new tool,  but before we have a whole discussion here that nobody saw I will open an "enhancement" issue :-)

**Reply by lowlander** [reply]
I opened an issue #29087 for this, so further discussions can take place there.
